### PR TITLE
port supplycamp to 1.11

### PR DESCRIPTION
**Pull request by Raycoms**
Closes #

# Changes proposed in this pull request:
-
-
-

Review please

### CODE REVIEW

**Comment by xavierh** [comment]
Tested, works

File: src/main/java/com/minecolonies/coremod/items/ItemSupplyCampDeployer.java [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+    }
+
+    private boolean checkAndPlaceSupplyCamp(final World world, @NotNull final BlockPos pos, @NotNull final EnumFacing direction)
+    {
+        EnumFacing facing = direction;
+        for (int i = 0; i < CHECK_X_DIRECTIONS; i++)
+        {
+            switch (facing)
+            {
+                case NORTH:
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/items/ItemSupplyCampDeployer.java:148]: MAJOR Reduce this switch case number of lines from 7 to at most 5, for example by extracting code into methods. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1151)

File: src/main/java/com/minecolonies/coremod/items/ItemSupplyCampDeployer.java [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+                    break;
+                case EAST:
+                    if (StructureWrapper.tryToLoadAndPlaceSupplyCampWithRotation(world, SUPPLY_CAMP_STRUCTURE_NAME,
+                            pos.add(OFFSET_SOUTH_EAST, 0, OFFSET_NORTH_EAST), Constants.ROTATE_0_TIMES, Mirror.NONE))
+                    {
+                        return true;
+                    }
+                    facing = EnumFacing.NORTH;
+                    break;
+                case WEST:
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/items/ItemSupplyCampDeployer.java:164]: MAJOR Reduce this switch case number of lines from 7 to at most 5, for example by extracting code into methods. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1151)

File: src/main/java/com/minecolonies/coremod/items/ItemSupplyCampDeployer.java [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+            {
+                case NORTH:
+                    if (StructureWrapper.tryToLoadAndPlaceSupplyCampWithRotation(world, SUPPLY_CAMP_STRUCTURE_NAME,
+                            pos.add(OFFSET_NORTH_EAST, 0, OFFSET_NORTH_WEST), Constants.ROTATE_THREE_TIMES, Mirror.NONE))
+                    {
+                        return true;
+                    }
+                    facing = EnumFacing.WEST;
+                    break;
+                case EAST:
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/items/ItemSupplyCampDeployer.java:156]: MAJOR Reduce this switch case number of lines from 7 to at most 5, for example by extracting code into methods. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1151)

File: src/main/java/com/minecolonies/coremod/items/ItemSupplyCampDeployer.java [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+                    break;
+                case WEST:
+                    if (StructureWrapper.tryToLoadAndPlaceSupplyCampWithRotation(world, SUPPLY_CAMP_STRUCTURE_NAME,
+                            pos.add(OFFSET_NORTH_WEST, 0, OFFSET_SOUTH_WEST), Constants.ROTATE_TWICE, Mirror.NONE))
+                    {
+                        return true;
+                    }
+                    facing = EnumFacing.SOUTH;
+                    break;
+                default:
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/items/ItemSupplyCampDeployer.java:172]: MAJOR Reduce this switch case number of lines from 6 to at most 5, for example by extracting code into methods. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1151)

File: src/main/java/com/minecolonies/coremod/util/StructureWrapper.java [hunk 5]
```diff
@@ -162,6 +194,46 @@ public void rotate(final int times, @NotNull final World world, @NotNull final B
...
+        }
+        return true;
+    }
+
+    /**
+     * Place a structure into the world.
+     *
+     * @param pos coordinates
+     */
     private void placeStructure(@NotNull final BlockPos pos)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/util/StructureWrapper.java:237]: CRITICAL Refactor this method to reduce its Cognitive Complexity from 24 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 80 issues
* ![CRITICAL][CRITICAL] 13 critical
* ![MAJOR][MAJOR] 22 major
* ![MINOR][MINOR] 44 minor
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [AbstractEntityAIStructure.java#L216](https://github.com/Minecolonies/minecolonies/blob/7759937a194d3dde4a3fef23a837642f596e52a7/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java#L216): Refactor this method to reduce its Cognitive Complexity from 35 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![CRITICAL][CRITICAL] [AbstractEntityAIStructure.java#L316](https://github.com/Minecolonies/minecolonies/blob/7759937a194d3dde4a3fef23a837642f596e52a7/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java#L316): Reduce the number of conditional operators (6) used in the expression (maximum allowed 5). [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1067)
1. ![CRITICAL][CRITICAL] [AbstractEntityAIStructure.java#L351](https://github.com/Minecolonies/minecolonies/blob/7759937a194d3dde4a3fef23a837642f596e52a7/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java#L351): The Cyclomatic Complexity of this method "structureStep" is 20 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![CRITICAL][CRITICAL] [AbstractEntityAIStructure.java#L351](https://github.com/Minecolonies/minecolonies/blob/7759937a194d3dde4a3fef23a837642f596e52a7/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java#L351): Refactor this method to reduce its Cognitive Complexity from 26 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![CRITICAL][CRITICAL] [AbstractEntityAIStructure.java#L362](https://github.com/Minecolonies/minecolonies/blob/7759937a194d3dde4a3fef23a837642f596e52a7/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java#L362): Reduce the number of conditional operators (7) used in the expression (maximum allowed 5). [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1067)
1. ![CRITICAL][CRITICAL] [AbstractEntityAIStructure.java#L574](https://github.com/Minecolonies/minecolonies/blob/7759937a194d3dde4a3fef23a837642f596e52a7/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java#L574): Refactor this method to reduce its Cognitive Complexity from 21 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![CRITICAL][CRITICAL] [AbstractEntityAIStructure.java#L624](https://github.com/Minecolonies/minecolonies/blob/7759937a194d3dde4a3fef23a837642f596e52a7/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java#L624): Reduce the number of conditional operators (7) used in the expression (maximum allowed 5). [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1067)
1. ![CRITICAL][CRITICAL] [AbstractEntityAIStructure.java#L879](https://github.com/Minecolonies/minecolonies/blob/7759937a194d3dde4a3fef23a837642f596e52a7/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java#L879): The Cyclomatic Complexity of this method "placeBlock" is 41 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![CRITICAL][CRITICAL] [AbstractEntityAIStructure.java#L879](https://github.com/Minecolonies/minecolonies/blob/7759937a194d3dde4a3fef23a837642f596e52a7/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java#L879): Refactor this method to reduce its Cognitive Complexity from 43 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![CRITICAL][CRITICAL] [AbstractEntityAIStructure.java#L1013](https://github.com/Minecolonies/minecolonies/blob/7759937a194d3dde4a3fef23a837642f596e52a7/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java#L1013): The Cyclomatic Complexity of this method "spawnEntity" is 17 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

[CRITICAL]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Comment by marvin-bitterlich** [comment]
LGTM

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/811/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

**Pull request merged**